### PR TITLE
Feature/merge r36 develop

### PR DESCRIPTION
**Pull request by CuriousVini**
- AuthorizationTest - kept changes from release/3.6 - removed import co.cask.cdap.internal.test.AppJarHelper; because it was causing compilation error
- StreamStateStoreUpgrader - kept changes from release/3.6
- ExistingEntitySystemMetadataWriter.java - kept changes from release/3.6 and remove conflicts for DatasetId and NamespaceId
- DatasetUpgrader - kept changes from release/3.6 except conflicts for NamespaceId
- HBaseQueueAdmin - kept changes from release/3.6
- MapReduceContainerLauncher.java- kept changes from release/3.6 
- IntegrationTestBase.java - Kept changes from develop

Build: http://builds.cask.co/browse/CDAP-DUT4968-5

### CODE REVIEW

**Comment by anwar6953** [comment]
I don't think we want to unconditionally take the changes from develop for MapReduceContainerLauncher.
For instance, this fix went into develop: https://github.com/caskdata/cdap/commit/254848355ef28c195508a20bf704674959993885
And this fix went into release/3.5 and release/3.6:
https://github.com/caskdata/cdap/pull/6980

We want to make sure we don't miss out on either of the two fixes.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java [hunk 1]
```diff
@@ -89,12 +91,15 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         builder.put(entry.getKey(), jobDataMap.getString(entry.getKey()));
       }
 
+      ProgramId programId = new ProgramId(namespaceId, applicationId, programType, programName);
```
Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java]: should use `ProgramId programId = new ApplicationId(namespaceId, applicationId, appVersion).program(programType, programId)` to include appVersion

Comment (CuriousVini) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java]: done

**Comment by CuriousVini** [comment]
@anwar6953 yea, we have merged changes from both the commits. I have rebased this branch because there were merge conflicts:

cdap-cli/bin/cdap-cli.bat - merge conflicts, kept changes from develop
cdap-standalone/bin/cdap.bat - merge conflicts, kept changes from develop

**Comment by bdmogal** [comment]
@CuriousVini why were there conflicts in the `.bat` files? Did anything change on `release/3.5`/`release/3.6`?

**Comment by CuriousVini** [comment]
@bdmogal Its because of https://github.com/caskdata/cdap/pull/6774/files PR got merged to develop

**Comment by bdmogal** [comment]
But how come there were conflicts? did something change in release/3.5 or 3.6?

**Comment by bdmogal** [comment]
nvm. this is probably it - https://github.com/caskdata/cdap/pull/6988/commits/8921e56c4f66cf10be8be76e5e3ba1fedbcb63ce

File: cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java [hunk 2]
```diff
@@ -142,14 +148,21 @@ private void writeSystemMetadataForDatasets(NamespaceId namespace, DatasetFramew
     SystemDatasetInstantiatorFactory systemDatasetInstantiatorFactory =
       new SystemDatasetInstantiatorFactory(locationFactory, dsFramework, cConf);
     try (SystemDatasetInstantiator systemDatasetInstantiator = systemDatasetInstantiatorFactory.create()) {
+      UserGroupInformation ugi = impersonator.getUGI(namespace.toId().toEntityId());
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java]: this seems redundant:
`namespace.toId().toEntityId()`.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java [hunk 3]
```diff
@@ -28,6 +30,7 @@
 import co.cask.cdap.data2.util.hbase.HTableNameConverter;
 import co.cask.cdap.data2.util.hbase.HTableNameConverterFactory;
 import co.cask.cdap.proto.DatasetSpecificationSummary;
+import co.cask.cdap.proto.NamespaceMeta;
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java:33]: Shouldn't this be moved down by 1 line?
Else, it'll break checkstyle, right?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java [hunk 4]
```diff
@@ -94,16 +97,21 @@
...
+  private final Impersonator impersonator;
 
   @Inject
-  HBaseQueueAdmin(Configuration hConf,
-                  CConfiguration cConf,
-                  LocationFactory locationFactory,
-                  HBaseTableUtil tableUtil,
-                  DatasetFramework datasetFramework,
-                  TransactionExecutorFactory txExecutorFactory) {
+  public HBaseQueueAdmin(Configuration hConf,
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java]: leave the `public` modifier out?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java (hunk 3 above)
Comment (CuriousVini) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java:33]: @anwar6953 There is no checkstyle error. I built the changes locally. The build passed.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java (hunk 2 above)
Comment (CuriousVini) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java]: @anwar6953 `namespace.toId()` provides Id.Namespace and `namespace.toId().toEntityId()` provides NamesapceId which is needed for `impersonator.getUGI()`

Comment (CuriousVini) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java]: Oh, my bad. Thats correct we only need `namespace`

**Comment by CuriousVini** [comment]
@anwar6953 Addressed comments. Please take a look when you get a chance

**Comment by anwar6953** [comment]
One last concern. Be good if @chtyim confirms the changes referenced here:
https://github.com/caskdata/cdap/pull/6988#issuecomment-255815301

**Comment by sreevatsanraman** [comment]
Build passed. @anwar6953 @chtyim can we have a LGTM from one of you?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
    * Launches the given main class. The main class will be loaded through the {@link MapReduceClassLoader}.
    *
    * @param mainClassName the main class to launch
-   * @param args arguments for the main class
+   * @param args          arguments for the main class
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java:43]: nit - seems like extra space was brought into the javadoc during conflict resolution

Comment (anwar6953) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java:43]: actually, it seems intentional, to make the description of the param align with the description of the param above it.

**Comment by anwar6953** [comment]
LGTM

**Comment by CuriousVini** [comment]
Build is green. Merging.

**Pull request merged**